app/controllers/api/v1/salesman_controller.rb

```diff
@@ -7,34 +7,39 @@ def create
           result = salesman_service
 
           if result.errors.blank?
-            render status: 201, json: { data: result, status: 201 }
+            render status: 201, json: {data: result, status: 201}
           else
-            render nothing: true, status: 400, json: { status: 400, data: result.errors[:name] }
+            render nothing: true, status: 400, json: {status: 400, data: result.errors[:name]}
           end
         else
-          render nothing: true, status: 400, json: { status: 400, data: 'Bad Request' }
+          render nothing: true, status: 400, json: {status: 400, data: 'Bad Request'}
         end
       end
 
       def update
-        salesman = ::Salesman.find_by(id: params[:id])
+        salesman = salesman_service_update
+
+        if salesman == 404
```

Comment:
Salesman service shouldn't be responsible to handle with status code, this responsibility belongs to controller. There can return nil, raise an exception or other returns.

Reply:
good observation